ev3dev2/motor.py

```diff
@@ -2237,30 +2224,103 @@ def follow_gyro_angle(self,
+        Use a GyroSensor to rotate in place for ``target_angle``
 
         ``speed`` is the desired speed of the midpoint of the robot
 
-        ``target_angle`` is the target angle we want to pivot to
+        ``target_angle`` is the number of degrees we want to rotate
+
+        ``brake`` hit the brakes once we reach ``target_angle``
 
         ``wiggle_room`` is the +/- angle threshold to control how accurate the turn should be
 
         ``sleep_time`` is how many seconds we sleep on each pass through
             the loop.  This is to give the robot a chance to react
             to the new motor settings. This should be something small such
             as 0.01 (10ms).
+        """
+
+        # MoveTank does not have information on wheel size and distance (that is
+        # MoveDifferential) so we must use a GyroSensor to control how far we rotate.
+        assert self._gyro, "GyroSensor must be defined"
+
+        if not target_angle:
+            return
+
+        speed_native_units = speed.to_native_units(self.left_motor)
+        init_angle = self._gyro.angle
+        first = True
+
+        while True:
+            current_angle = self._gyro.angle
+            degrees_moved = abs(current_angle - init_angle)
+            delta = abs(abs(target_angle) - degrees_moved)
+
+            if delta <= wiggle_room:
+                self.stop(brake=brake)
+                break
+
+            # our goal is to rotate target_angle clockwise
+            if target_angle > 0:
+
+                # we have not gone far enough, rotate clockwise
+                if first or degrees_moved < target_angle:
+                    left_speed = SpeedNativeUnits(speed_native_units)
+                    right_speed = SpeedNativeUnits(-1 * speed_native_units)
+
+                # we went too far, rotate counter-clockwise
+                else:
+                    left_speed = SpeedNativeUnits(-1 * speed_native_units)
+                    right_speed = SpeedNativeUnits(speed_native_units)
+
+            # our goal is to rotate target_angle counter-clockwise
+            else:
+
+                # we have not gone far enough, rotate counter-clockwise
+                if first or degrees_moved < abs(target_angle):
+                    left_speed = SpeedNativeUnits(-1 * speed_native_units)
+                    right_speed = SpeedNativeUnits(speed_native_units)
+
+                # we went too far, rotate clockwise
+                else:
+                    left_speed = SpeedNativeUnits(speed_native_units)
+                    right_speed = SpeedNativeUnits(-1 * speed_native_units)
+
+            first = False
+            self.on(left_speed, right_speed)
+
+            if sleep_time:
+                time.sleep(sleep_time)
+
+    def turn_right(self, speed, degrees, brake=True, wiggle_room=2, sleep_time=0.01):
+        """
+        Rotate clockwise `degrees` in place
+        """
+        self._turn(speed, abs(degrees), brake, wiggle_room, sleep_time)
```

Comment:
It is just a safety check so that if the user does `turn_right(degrees=-15)` we still rotate in the correct direction.  We have these sorts of safety checks in other places so seems good to have them here for consistency.